app/models/application_choice.rb

```diff
@@ -11,6 +11,8 @@ class ApplicationChoice < ApplicationRecord
   has_one :provider, through: :course
   has_one :accredited_provider, through: :course, class_name: 'Provider'
 
+  has_many :notes, depenant: :destroy
```

Comment:
Typo here, but we don't need `dependent: :destroy` because by default we do a "cascading delete" in the database. You can [see it in action in this spec](https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training/blob/master/spec/models/candidate_spec.rb#L15-L35). It's probably worth adding an extra test for these models to that spec to make sure it does what we want.

Reply:
This was tripping a rubocop so I have added a global disable for Disable `Rails/HasManyOrHasOneDependent`.

Reply:
Having a cascading delete constraint in the DB is a good thing. Not sure whether we should have `dependent: destroy` is a bad thing though. We have lots of `presence: true` validations in Rubyland backed by `NOT NULL` constraints in the DB. It made me wonder what happens to the models when a cascading delete is triggered in the DB - does the collection of dependent models in memory get out of sync with the DB?

Reply:
🤔 good point - have had a bit of a google around and haven't seen anything that would explain what is happening here.